File: 프로그래머스/2/42586. 기능개발/기능개발.js
function makeDays(progresses, speeds) {
    const days = progresses.map((progress, i) => Math.ceil((100 - progress) / speeds[i]));
    return days;
}

function makeAnswer (days) {
    const answer = [];
    let standardDay = days[0];
    let count = 1;
    
    for (let i = 1; i < days.length; i++) {
        if (standardDay >= days[i]) {
            count++;
        } else {
            answer.push(count);
            count = 1;
            standardDay = days[i];
        }
    }
    
    answer.push(count);
    return answer;
}

function solution(progresses, speeds) {
    const days = makeDays(progresses, speeds);
    const answer = makeAnswer(days);   
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    queue = deque()
    
    # 배포일 계산
    for index, current_progress in enumerate(progresses):
        current_speed = speeds[index]
        time = (100 - current_progress) // current_speed
        if (100 - current_progress) % current_speed:
            time += 1
        queue.append(time)
    
    while queue:
        target = queue.popleft()
        count = 1
        while queue and queue[0] <= target:
            count += 1
            queue.popleft()
        answer.append(count)
    return answer